core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1159,6 +1177,48 @@ public RepositoriesService getRepositoriesService() {
     }
 
     /**
+     * Resolve the {@link SnapshotId} for the given {@link Snapshot}.
+     *
+     * @param repositoryName repository name
+     * @param snapshotName snapshot name
+     * @return the snapshot id
+     * @throws SnapshotMissingException if the snapshot could not be found
+     */
+    public SnapshotId resolveSnapshotId(final String repositoryName, final String snapshotName) {
+        final Snapshot snapshot = new Snapshot(repositoryName, snapshotName);
+        validate(snapshot);
+        List<SnapshotId> ids = resolveSnapshotIds(repositoryName, Collections.singleton(snapshotName));
+        assert ids.size() == 1; // if the id wasn't found, a SnapshotMissingException should've be thrown
+        return ids.get(0);
+    }
+
+    /**
+     * Resolve the {@link SnapshotId}s for the given {@link Snapshot}s.
+     *
+     * @param repositoryName repository name
+     * @param snapshotNames snapshot names
+     * @return the snapshot ids
+     * @throws SnapshotMissingException if a snapshot could not be found
+     */
+    public List<SnapshotId> resolveSnapshotIds(final String repositoryName, final Set<String> snapshotNames) {
```

Comment:
This method is only used by the single collection version above. Do you see other usages?